types/react-native/test/animated.tsx

```diff
@@ -164,6 +164,27 @@ function TestAnimatedAPI() {
             <ForwardComp ref={ForwardCompRef} width={1} />
             <AnimatedForwardComp ref={AnimatedForwardCompRef} width={10} />
             <Animated.Image style={position.getTranslateTransform()} source={{ uri: 'https://picsum.photos/200' }} />
+
+            {/* AnimatedComponent function prop args should infer original types */}
+            <Animated.View
+              style={{opacity: v1}}
+              //should infer (event: LayoutChangeEvent) => void;
+              onLayout={e => console.log(e.nativeEvent.layout)}
```

Comment:
Because we specifically want to know that these args are not `any`, it might be better to expect an exact type, e.g.:

```suggestion
              onLayout={e => {
                e; // $ExpectType (event: LayoutChangeEvent) => void
              }}
```

That is, if that works at all; I don’t think I’ve tried it in a callback function before.